packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -613,32 +606,38 @@ class _RawAutocompleteOptionsState extends State<_RawAutocompleteOptions> {
 
   @override
   Widget build(BuildContext context) {
-    return CompositedTransformFollower(
-      link: widget.optionsLayerLink,
-      followerAnchor: switch (widget.optionsViewOpenDirection) {
-        OptionsViewOpenDirection.up => Alignment.bottomLeft,
-        OptionsViewOpenDirection.down => Alignment.topLeft,
-      },
-      // When the field goes offscreen, don't show the options.
-      showWhenUnlinked: false,
-      child: CustomSingleChildLayout(
-        delegate: _RawAutocompleteOptionsLayoutDelegate(
-          fieldSize: widget.optionsLayerLink.leaderSize!,
-          fieldOffset: fieldOffset,
-          optionsViewOpenDirection: widget.optionsViewOpenDirection,
-          textDirection: Directionality.of(context),
-        ),
-        child: TextFieldTapRegion(
-          child: AutocompleteHighlightedOption(
-            highlightIndexNotifier: widget.highlightIndexNotifier,
-            // optionsViewBuilder must be able to look up
-            // AutocompleteHighlightedOption in its context.
-            child: Builder(
-              builder: widget.builder,
+    // Wrap the options view in a LayoutBuilder so it can retrieve the most
+    // updated leader size.
+    return LayoutBuilder(
+      builder: (BuildContext context, BoxConstraints constraints) {
+        return CompositedTransformFollower(
+          link: widget.optionsLayerLink,
+          followerAnchor: switch (widget.optionsViewOpenDirection) {
+            OptionsViewOpenDirection.up => Alignment.bottomLeft,
+            OptionsViewOpenDirection.down => Alignment.topLeft,
+          },
+          // When the field goes offscreen, don't show the options.
+          showWhenUnlinked: false,
+          child: CustomSingleChildLayout(
+            delegate: _RawAutocompleteOptionsLayoutDelegate(
+              fieldSize: widget.optionsLayerLink.leaderSize!,
```

Comment:
Actually I think you probably don't need the layout builder. The size of the field needs to be evaluated when the options view does layout, so doing that in LayoutDelegate's implementation should be fine.